### DISCUSSION TITLE
How use Both Secure Boot  and OTA

### DISCUSSION THREAD

**Discussion by xshuoo** [espressif/arduino-esp32#11103] [Q&A]
### Board

esp32-s3

### Device Description

esp32-s3 chip only

### Hardware Configuration

no other HW

### Version

v3.0.6

### IDE Name

esp32-s3

### Operating System

win10

### Flash frequency

80Mhz

### PSRAM enabled

yes

### Upload speed

115200

### Description

There are some ESP-S3 chips that use internal flash. The firmware is quite small, but I’ve secured it with secure boot, and I really hope to update it in the future. However, I’m not sure how to configure the chip for that purpose.

### Sketch

```cpp
helllo world
```

### Debug Message

```plain
print hello world ok
```

### Other Steps to Reproduce

try to keep the fireware and OTA 

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by dzungpv** [comment]
Try using Arduino as ESP IDF and you can follow IDF guide here to enable secure boot and flash encryption, It is not available with Arduino IDE. https://docs.espressif.com/projects/esp-idf/en/v5.1.5/esp32s3/security/secure-boot-v2.html
